File: CardPeak.UI.ClientApp/src/services/api/transactionsController.ts
import axios from 'axios';

const API = {
    GET_TRANSACTIONS_BY_CLIENT: (clientName: string) => {
        return '/transactions/client/' + clientName;
    },
    GET_TRANSACTIONS_BY_BATCH: (batchId: number) => {
        return '/transactions/batch/' + batchId;
    },
    DELETE_TRANSACTION: (transactionId: number) => {
        return '/transactions/delete/' + transactionId;
    }
};

export function deleteTransaciton(
    id: number,
    deleteComplete: (id: number) => void,
    deleteError: (error: string) => void
) {
    axios
        .post(API.DELETE_TRANSACTION(id))
        .then(r => {
            deleteComplete(id);
        })
        .catch(reason => {
            deleteError(reason.message);
        });
}

export function getTransactionsByClient(
    clientName: string,
    successCallback: (
        data: CardPeak.Entities.ApprovalTransaction[]
    ) => void,
    errorCallback: (message: string) => void
) {
    axios
        .get(API.GET_TRANSACTIONS_BY_CLIENT(clientName))
        .then(r => {
            successCallback(
                r.data as CardPeak.Entities.ApprovalTransaction[]
            );
        })
        .catch(reason => {
            errorCallback(reason.message);
        });
}

export function getTransactionsByBatch(
    batchId: number,
    successCallback: (
        data: CardPeak.Entities.ApprovalTransaction[]
    ) => void,
    errorCallback: (message: string) => void
) {
    axios
        .get(API.GET_TRANSACTIONS_BY_BATCH(batchId))
        .then(r => {
            successCallback(
                r.data as CardPeak.Entities.ApprovalTransaction[]
            );
        })
        .catch(reason => {
            errorCallback(reason.message);
        });
}

File: CardPeak.WebAPI/Controllers/TransactionsController.cs
using CardPeak.Core.Service;
using CardPeak.Service;
using System.Web.Http;

namespace CardPeak.WebAPI.Controllers
{
    [RoutePrefix("api/transactions")]
    public sealed class TransactionsController : ApiController
    {
        private ITransactionService TransactionService { get; set; }

        public TransactionsController()
        {
            this.TransactionService = new TransactionService(new Repository.EF.CardPeakDbContext());
        }

        [HttpPost]
        [Route("delete/{id}")]
        public IHttpActionResult Delete(int id)
        {
            var result = this.TransactionService.DeleteTransaction(id);
            if (!result)
            {
                return this.NotFound();
            }

            return this.Ok(id);
        }

        [HttpGet]
        [Route("client/{clientName}")]
        public IHttpActionResult GetTransactionsByClient(string clientName)
        {
            var result = this.TransactionService.GetTransactionsByClient(clientName);
            if (result == null)
            {
                this.NotFound();
            }

            return this.Ok(result);
        }

        [HttpGet]
        [Route("batch/{id}")]
        public IHttpActionResult GetTransactionsByBatch(int batchId)
        {
            var result = this.TransactionService.GetTransactionsByBatch(batchId);
            if (result == null)
            {
                this.NotFound();
            }

            return this.Ok(result);
        }
    }
}
